src/com/javacore/lesson09/UserRegistration.java

```diff
@@ -0,0 +1,10 @@
+package com.javacore.lesson09;
+
+public class UserRegistration {
+
+    public static void main(String[] args) {
```

Comment:
better to keep user input here, not in User class
In your solution user class has several responsibilities - domain (user class by itself with construction, getters/setters and input data, data validation)
At least you can move data input from there to make it more clean.